command/scripting.go

```diff
@@ -0,0 +1,204 @@
+package command
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/meitu/titan/context"
+	"github.com/meitu/titan/encoding/resp"
+	"github.com/yuin/gopher-lua"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var cli *context.ClientContext
+
+func NewClientContext(ctx *Context) *context.ClientContext {
+	now := time.Now()
+	cli := &context.ClientContext{
+		Namespace:  ctx.Client.Namespace,
+		ID:         ctx.Client.ID,
+		RemoteAddr: ctx.Client.RemoteAddr,
+		DB:         ctx.Client.DB,
+		Created:    now,
+		Updated:    now,
+	}
+	return cli
+}
+
+const luaRedisTypeName = "redis"
+
+// Registers my redis type to given L.
+func registerRedisType(L *lua.LState) {
+	//
+	mt := L.NewTypeMetatable(luaRedisTypeName)
+	L.SetGlobal("redis", mt)
+	// static attributes
+	L.SetField(mt, "call", L.NewFunction(newRedisCall))
+}
+
+func parseLuaValue(ctx *Context, data lua.LValue) {
+	Type := data.Type().String()
+	if Type == "string" {
+		if lv, ok := data.(lua.LString); ok {
+			resp.ReplySimpleString(ctx.Out, string(lv))
+		}
+	} else if Type == "number" {
+		if intv, ok := data.(lua.LNumber); ok {
+			resp.ReplyInteger(ctx.Out, int64(intv))
+		}
+	} else if Type == "boolean" {
+		resp.ReplyNullBulkString(ctx.Out)
+	} else if Type == "nil" {
+		resp.ReplyNullBulkString(ctx.Out)
+	} else if Type == "table" {
+		var array []lua.LValue
+		a := data.(*lua.LTable)
+		len := 0
+		a.ForEach(func(value lua.LValue, value2 lua.LValue) {
+			len++
+			array = append(array, value2)
+		})
+		resp.ReplyArray(ctx.Out, len)
+		for _, r := range array {
+			parseLuaValue(ctx, r)
+		}
+	} else {
+		resp.ReplyNullBulkString(ctx.Out)
+	}
+}
+
+// Constructor
+func newRedisCall(L *lua.LState) int {
+	var rest []string
+	// filter data from lua to redis command without first arg
+	GetLuaTop := L.GetTop()
```

Comment:
What if GetLuaTop < 2 ?
Return an error when there is no arguments according to redis:
>ERR Error running script (call to f_0a907e1429221a4d85516cab7fd219a82a9439d8): @user_script:1: @user_script: 1: Please specify at least one argument for redis.call()